src/ucp/wireup/select.c

```diff
@@ -195,27 +196,91 @@ ucp_wireup_get_missing_amo_flag_desc_fop(uint64_t flags, uint64_t required_flags
     return ucp_wireup_get_missing_amo_flag_desc(flags, required_flags, op_size, 1, buf, len);
 }
 
-static int ucp_wireup_check_flags(const uct_tl_resource_desc_t *resource,
-                                  uint64_t flags, uint64_t required_flags,
-                                  const char *title, const char ** flag_descs,
-                                  char *reason, size_t max)
+static void ucp_wireup_init_extended_flags(ucp_wireup_extended_flags_t *flags)
+{
+    flags->mandatory = 0;
+    flags->optional  = 0;
+}
+
+static void ucp_wireup_format_missing_opt_flags_msg(char *missing_flags_str,
+                                                    size_t flags_str_max_size,
```

Comment:
better pass string buffer here (as last parameter), and avoid ucs_snprintf_safe (use string buffer function)
string buffer should also have API to print flags to the string and macro to init on stack